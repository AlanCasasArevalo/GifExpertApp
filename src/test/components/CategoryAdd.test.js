import {shallow} from "enzyme";
import React from "react";
import AddCategory from "../../components/AddCategory";

describe('AddCategory Component test', () => {

    const setCategories = () => {
    }
    let wrapper = shallow(
        <AddCategory
            setCategories={setCategories}
        />)

    beforeEach(() => {
        wrapper = shallow(
            <AddCategory
                setCategories={setCategories}
            />)
    })

    test('AddCategory should be defined', () => {
        expect(<AddCategory
            setCategories={setCategories}
        />).toBeDefined()
    })
    test('AddCategory should be an object', () => {
        expect(typeof <AddCategory
            setCategories={setCategories}
        />).toBe('object')
    })

    test('AddCategory should return a snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('form should get input', () => {
        const form = wrapper.find('form')
        const {children} = form.props()
        expect(children.type).toStrictEqual('input')
    })
    
})






