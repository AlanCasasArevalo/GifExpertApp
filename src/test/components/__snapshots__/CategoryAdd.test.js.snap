// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Component test AddCategory should return a snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
